addons/account/models/account.py

```diff
@@ -690,50 +691,91 @@ def compute_all(self, price_unit, currency=None, quantity=1.0, product=None, par
-        else:
-            total_excluded, total_included, base = base_values
-
         # Sorting key is mandatory in this case. When no key is provided, sorted() will perform a
         # search. However, the search method is overridden in account.tax in order to add a domain
         # depending on the context. This domain might filter out some taxes from self, e.g. in the
         # case of group taxes.
-        for tax in self.sorted(key=lambda r: r.sequence):
+        sorted_taxes = self.sorted(key=lambda r: r.sequence)
+
+        base_value = self.env.context.get('base_value')
+        if base_value:
+            base = base_value
+        else:
+            base = round(price_unit * quantity, prec)
+
+        # For the computation of move lines, we could have a negative base value.
+        # In this case, compute all with positive values and negative them at the end.
+        if base < 0:
+            base = -base
+            sign = -1
+        else:
+            sign = 1
+
+        # Compute the new base amount based on included fixed/percent amount.
+        def compute_base(base_amount, fixed_amount, percent_amount):
+            if fixed_amount == 0.0 and percent_amount == 0.0:
+                return base_amount
+            return round((base_amount - fixed_amount) / (1.0 + percent_amount / 100.0), prec)
+
+        # 1) Iterate the taxes in the reversed sequence order to retrieve the initial base of the computation.
+        #     tax  |  base  |  amount  |
+        # /\ ----------------------------
+        # || tax_1 |  XXXX  |          | <- we are looking for that, it's the total_excluded
+        # || tax_2 |        |          |
+        # || tax_3 |        |          |
+        # ||  ...  |   ..   |    ..    |
+        #    ----------------------------
+        incl_fixed_amount = incl_percent_amount = 0
+        for i in range(len(sorted_taxes) - 1, -1, -1):
+            tax = sorted_taxes[i]
+            if tax.include_base_amount:
+                base = compute_base(base, incl_fixed_amount, incl_percent_amount)
+                incl_fixed_amount = incl_percent_amount = 0
+            if tax.price_include:
+                if tax.amount_type == 'fixed':
+                    incl_fixed_amount += tax.amount
+                elif tax.amount_type == 'percent':
+                    incl_percent_amount += tax.amount
+        total_excluded = total_included = base = compute_base(base, incl_fixed_amount, incl_percent_amount)
+
+        # 2) Iterate the taxes in the sequence order to fill missing base/amount values.
+        #      tax  |  base  |  amount  |
+        # ||  ----------------------------
+        # ||  tax_1 |   OK   |   XXXX   |
+        # ||  tax_2 |  XXXX  |   XXXX   |
+        # ||  tax_3 |  XXXX  |   XXXX   |
+        # \/  ...  |   ..   |    ..    |
+        #     ----------------------------
+        for tax in sorted_taxes:
```

Comment:
The `with_context(force_price_include=False)` seems necessary to me regarding the provided example...